scripts/document_updater.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import re
+
+from typing import List
+from dataclasses import dataclass
+from pathlib import Path
+from inspect import cleandoc
+
+from exasol_advanced_analytics_framework.example import generator as example_generator
+
+
+class ParseException(Exception):
+    """
+    If input file is not well-formed.
+    """
+
+
+@dataclass
+class Template:
+    path: str
+
+    def render(self):
+        if self.path != "example/generator.py":
+            raise ParseException("document_updater.Template currently only"
+                                 " supports path example/generator.py")
+        return "\n".join([
+            "<!-- The example is deliberately tagged as language python since the major",
+            "     parts are in python. Formally, however, the python code is embedded into",
+            "     an SQL statement, though. -->",
+            "```python",
+            example_generator.create_script(),
+            "",
+            example_generator.execute_script() + ";",
+            "```",
+            ""])
+
+
+class ChunkReader:
+    """
+    Enables to replace chunks of a string by generated text, e.g. from
+    jinja templates.
```

Comment:
The class name and the description do not provide a good explanation of what this class is doing. Maybe a usage example can help.

Reply:
See next push